### DISCUSSION TITLE
Get text only changes

### DISCUSSION THREAD

**Discussion by ChizobaAmadi** [fleather-editor/fleather#427] [General]
Hello,
I'm implementing an auto-save feature by listening for text changes. Initially, I used `controller.addListener`, but I noticed it also triggers events for text selection and focus changes. I understand this behaviour is consistent with the native text field, so it's not a bug.

I then discovered` controller.document.changes`, which meets my needs for detecting only content changes. However, there seems to be limited documentation on this approach, so I wanted to check if there are any caveats I should be aware of.

Thank you

**Comment by amantoux** [comment]
Hello @ChizobaAmadi,
Yes indeed with `document.changes` you only get the actual changes made to the `Parchment` backing the editor

**Reply by ChizobaAmadi** [reply]
Ok,thanks for clarifying
